src/main/java/org/akaza/openclinica/service/ValidateServiceImpl.java

```diff
@@ -231,12 +227,16 @@ public boolean isUserHas_CRC_INV_DM_DEP_DS_SM_RoleInSite(List<StudyUserRoleBean>
         return false;
     }
 
-    public boolean isUserHas_DM_MON_RoleInStudy(List<StudyUserRoleBean> userRoles, String studyOID){
+    public boolean isUserHas_DM_MON_RoleInStudy(List<StudyUserRoleBean> userRoles, String studyOID) {
         Study publicStudy = getPublicStudy(studyOID);
         for (StudyUserRoleBean userRole : userRoles) {
-             if ((userRole.getRole().equals(Role.MONITOR) && publicStudy.getStudyId() == userRole.getStudyId())
-                    || (userRole.getRole().equals(Role.COORDINATOR) && publicStudy.getStudyId() == userRole.getStudyId()))
-                return true;
+            for (Study currStudy : publicStudy.getStudies())
```

Comment:
If studyOID can either be studyOID or siteOID, wouldn't publicStudy.getStudies() not return anything when it is a siteOID?

Reply:
They are actually different OIDs. publicStudy.getStudies().<iterate>.getStudyId is returning a siteOID.

Reply:
publicStudy.getStudy() returns null.

Reply:
only when studyOID is for a study, right? What does it return when studyOID represents a site OID?

Reply:
> publicStudy.getStudy() returns null.

so that's not desirable, right?

Reply:
yeah. when its a siteOId, publicSite.getStudyId() returns siteOID and publicSite.getStudy().getStudyId() returns studyOID().

Reply:
ok, please update the logic to handle both scenarios like you have done in isUserHas_CRC_INV_DM_DEP_DS_SM_RoleInSite method

Reply:
ok